domino/domino.py

```diff
@@ -525,15 +525,41 @@ def collaborators_get(self):
         url = self._routes.collaborators_get()
         return self._get(url)
 
-    def collaborators_add(self, usernameOrEmail, message=""):
+    def get_user_id(self, username_or_email): 
+        url = self._routes.users_get()
+        response = self.request_manager.get(url)
+        users = response.json()
+        user_from_user_name = [x for x in users if ('userName' in x) and (x['userName'] == username_or_email)]
+        user_from_email = [x for x in users if ('email' in x) and (x['email'] == username_or_email)]
+        user_list = user_from_user_name + user_from_email
+        if len(user_list) == 0:
+            raise Exception(f"Could not find user: {username_or_email}")
+        user = user_list[0]
+        userId = user['id']
+        return userId
```

Comment:
Since this is a public-facing method (i.e., not just test code), I think we need to be sure what the correct response should be in the case that the user isn't found. Throwing a bare `Exception` is probably not the best choice. My guess is that returning `None` is more appropriate. @mpkouznetsov or @ddl-mmahmoud?

To save time on the PR, I would suggest replacing the method with this.
```python
def get_user_id(self, username_or_email):
    """
    Return the user ID of the the user account matching the given search criteria.

    Args:
        username_or_email (str): the username or email account of a user

    Return:
        user_id (str): the matching user ID, or None if not found
    """
    url = self._routes.users_get()
    response = self.request_manager.get(url)
    users = response.json()

    user_id = None
    for user in users:
        if username_or_email in (user['userName'], user['email']):
            user_id = user['id']
            break
    return user_id
```
I grabbed two default users from a recent deployment...
```python
users = [
  {
    "firstName": "integration-test",
    "lastName": "integration-test",
    "fullName": "integration-test integration-test",
    "userName": "integration-test",
    "email": "test-notifs+integration-test@dominodatalab.com",
    "avatarUrl": "",
    "id": "60a301d1049cc4002e81c7c8"
  },
  {
    "firstName": "Jira",
    "lastName": "User",
    "fullName": "Jira User",
    "userName": "jira_user_anonymous",
    "email": "jira@dominodatalab.com",
    "avatarUrl": "",
    "id": "60a301e14d3bd8efde3eb94b"
  }
]
```
...then tested the following conditions, and they all seem to pass.
```python
# Test search by valid email
u0 = get_user_id(None, 'test-notifs+integration-test@dominodatalab.com')
assert u0 == '60a301d1049cc4002e81c7c8'

# Test search by valid username
u1 = get_user_id(None, 'jira_user_anonymous')
assert u1 == '60a301e14d3bd8efde3eb94b'

# Test non-existent user
u2 = get_user_id(None, 'foo')
assert u2 == None

# Test username substring does not return false positive
u3 = get_user_id(None, 'integra')
assert u3 == None

# Test email substring does not return false positive
u4 = get_user_id(None, '@dominodatalab.com')
assert u4 == None
```

Reply:
fixed as suggested.